dask_cuda/worker_spec.py

```diff
@@ -81,10 +84,14 @@ def worker_spec(
        'preload_argv': ['--create-cuda-context']}}}
 
     """
-    if (
-        enable_tcp_over_ucx or enable_infiniband or enable_nvlink
-    ) and protocol != "ucx":
-        raise TypeError("Enabling InfiniBand or NVLink requires protocol='ucx'")
+    if enable_tcp_over_ucx or enable_infiniband or enable_nvlink:
+        try:
+            _get_active_ucx_implementation_name(protocol)
+        except ValueError:
+            raise TypeError(
+                "Enabling InfiniBand or NVLink requires protocol='ucx', "
+                "protocol='ucxx' or protocol='ucx-old'"
+            )
```

Comment:
```suggestion
            raise TypeError(
                "Enabling InfiniBand or NVLink requires protocol='ucx', "
                "protocol='ucxx' or protocol='ucx-old'"
            ) from None
```

You probably want a `from None` here, to ensure that the `ValueError` doesn't end up in the traceback with the "while handling ... another exception was raised."

This is one of the reasons I don't like using exceptions for control flow like this, but that's not a big deal.